Rendera/include/cube.hpp

```diff
@@ -0,0 +1,209 @@
+    Material m_material;
+
+public:
+    Cube(vec3 position, vec3 a, vec3 b, vec3 c, Material mat) : m_position{position}, m_a{a}, m_b{b}, m_c{c}, m_material{mat} {};
+
+    ray normal(const vec3 &point)
+    {
+        vec3 p_pos, np_pos;
+        p_pos = point - m_position;
+        np_pos = point - m_a - m_b - m_c - m_position;
+        vec3 normal;
+        if (dot(p_pos, cross(m_a, m_b)) < 1e5)
+        {
+            normal = cross(m_a, m_b);
+            if (dot(m_c, normal) < 0)
+            {
+                normal = -normal;
+            }
+        }
+        else if (dot(p_pos, cross(m_a, m_c)) < 1e5)
+        {
+            normal = cross(m_a, m_c);
+            if (dot(m_b, normal) < 0)
+            {
+                normal = -normal;
+            }
+        }
+        else if (dot(p_pos, cross(m_b, m_c)) < 1e5)
+        {
+            normal = cross(m_b, m_c);
+            if (dot(m_a, normal) < 0)
+            {
+                normal = -normal;
+            }
+        }
+        else if (dot(np_pos, cross(m_a, m_b)) < 1e5)
+        {
+            normal = cross(m_a, m_b);
+            if (dot(-m_c, normal) < 0)
+            {
+                normal = -normal;
+            }
+        }
+        else if (dot(np_pos, cross(m_a, m_c)) < 1e5)
+        {
+            normal = cross(m_a, m_c);
+            if (dot(-m_b, normal) < 0)
+            {
+                normal = -normal;
+            }
+        }
+        else if (dot(np_pos, cross(m_b, m_c)) < 1e5)
+        {
+            normal = cross(m_b, m_c);
+            if (dot(-m_a, normal) < 0)
+            {
+                normal = -normal;
+            }
+        }
+        else
+        {
+            normal = vec3(0, 0, 0);
+        }
+        return ray(point, normal);
+    }
+
+    Material mat() const
+    {
+        return m_material;
+    }
+
+    vec3 position() const
+    {
+        return m_position;
+    }
+
+    vec3 a() const
+    {
+        return m_a;
+    }
+
+    vec3 b() const
+    {
+        return m_b;
+    }
+
+    vec3 c() const
+    {
+        return m_c;
+    }
+
+    HitInfo plane_intersect(const vec3 &n, const vec3 &p, const ray &r)
+    {
+        // if(dot(r.get_direction(),n) < 1e5 || dot(r.get_direction(),n) > -1e5){
+        //     return HitInfo(false, vec3(0,0,0), m_material, r, 0);
+        // }
+        // float t = dot(n,p - r.get_origin()) / dot(n,r.get_direction());
+        // if(t < 0){
+        //     return HitInfo(false, vec3(0,0,0), m_material, r, 0);
+        // }
+        // else{
+        //     return HitInfo(true, r.get_origin() + t * r.get_direction(), m_material, r, 0);
+        // }
+
+        float num = dot(n, p - r.get_origin());
+        if (abs(dot(r.get_direction(), n)) > 1e-5)
+        {
+            num = num / dot(r.get_direction(), n);
+        }
+        else
+        {
+            num *= INFINITY;
+        }
+
+        return HitInfo(true, r.get_origin() + num * r.get_direction(), m_material, r, 0);
```

Comment:
In modern C++, we prefer to use [braced-init-lists in return statements](https://stackoverflow.com/q/65457302/12591388).